exchanges/subscription/template.go

```diff
@@ -72,8 +73,28 @@ func (l List) ExpandTemplates(e IExchange) (List, error) {
 		expanded, err2 := expandTemplate(e, s, maps.Clone(ap), assets)
 		if err2 != nil {
 			err = common.AppendError(err, fmt.Errorf("%s: %w", s, err2))
-		} else {
-			subs = append(subs, expanded...)
+			continue
+		}
+
+		subs = append(subs, expanded...)
+	}
+
+	// Validate the subscriptions after expansion to capture fields that will be used in the template
+	if v, ok := e.(ListValidator); ok {
+		// Need to check against the already stored subscriptions, as we add additional subscriptions
+		storedSubs, err := e.GetSubscriptions()
+		if err != nil {
+			return nil, err
+		}
+		// Expand stored subscriptions to ensure we have the same template applied to validate
+		// against the incoming subscriptions. Pass e using IExchange interface to prevent exposure
+		// to ListValidator and avoid recursion.
+		expStoredSubs, err := storedSubs.ExpandTemplates(struct{ IExchange }{e})
+		if err != nil {
+			return nil, err
+		}
+		if err := v.ValidateSubscriptions(append(subs, expStoredSubs...)); err != nil {
```

Comment:
🚧 My concern with this is that it shouldn't be the job of ExpandTemplates to expand other subscriptions not passed in.
Was there a use-case for needing this, as opposed to hypothetically ?
If possible, I'd like to just use the subscriptions as they are.
I can't think of a use-case otherwise for how they got into subs without going through expand anyway.
🐒 I think `slices.Concat` is probably wiser because it doesn't imply touching `subs` underlying array
```suggestion
		if err := v.ValidateSubscriptions(slices.Concat(subs, storedSubs)); err != nil {
```

Reply:
It was mostly hypothetical, and can be seen as redundant. It's really only for when a sub not expanded gets inserted on error or intentionally. It forces consistency between the two in kind lists and doesn't assume anything. It should just skip anyway if its already expanded? If this is a real pinch point I can revert it; I see `ValidateSubscriptions` as being removed from ExpandTemplates eventually and this will force a better standard comparison. 

As for the `slices.Concat` change, this is predicated that subs cap is >= (subs len + storedSubs len) and you want a forced allocation. For this to leak you going to have to change the subs len. I don't think this is necessary even if we write temp to its array. If you are wondering about its memory foot print its already allocated as such until its collected. So if it didn't get validated it would still have the same cap size anyway.

Reply:
It's not a hot path anyway so I pushed that slices.Concat change, it won't hurt it. 🤷

Reply:
@gbjk Needs your input when you get a chance 🙇

Reply:
Sorry for delay.
Yes, I think I'm sticking to "please revert".
Less code is better, when it's not necessary.

Reply:
Reverted: [6654440](https://github.com/thrasher-corp/gocryptotrader/pull/1863/commits/6654440c1adb3f3e08145a16cd206dc4a7494b2b)